tests/jballerina-unit-test/src/test/resources/test-src/typedefs/type-definitions.bal

```diff
@@ -338,6 +338,29 @@ public function returnTupleWithSingletonType2() returns X2 {
     return ["x", 2, ()];
 }
 
+type Data record {|
+    string value;
+|};
+
+const annotation Data dataAnon on type;
+
+@dataAnon {
+    value: "T1"
+}
+type Person2 record {|
+    string name;
+|};
+
+function testAnnotWithRecordTypeDefinition() {
+    Person2 foo = {name: "James"};
+    typedesc<any> t = typeof foo;
+    Data? annon = t.@dataAnon;
+    assertEquality("{\"value\":\"T1\"}", annon.toString());
+
+    Data data = {value: "T2"};
+    assertEquality("{\"value\":\"T2\"}" , data.toString());
```

Comment:
Is this to test the availability of Data at runtime? If so, aren't tests that actually validate against `Data` better? E.g., positive and negative `is` checks. Can also use a record with a default, use cloneWithType with it with a value that doesn't have a field for the field with the default, and test that the default is used.